### PR TITLE
Upgrade diktat to 1.2.0

### PR DESCRIPTION
**Pull request by nulls**
Bump default `diktat` version to latest `1.1.0` -> `1.2.0`

It closes https://github.com/saveourtool/diktat/issues/1399

### CODE REVIEW

**Comment by nulls** [comment]
Hi,

We (diktat) released the new version `1.2.0`. It has breaking changes and requires changes in stopless.
This PR with supporting these changes.

**Comment by akuleshov7** [comment]
Probably there can be some problems with backward compatibility. Is there any chance to raise an error for older versions of diktat?

File: lib/src/diktat/java/com/diffplug/spotless/glue/diktat/DiktatFormatterFunc.java [hunk 1]
```diff
@@ -33,15 +33,13 @@
 public class DiktatFormatterFunc implements FormatterFunc.NeedsFile {
 
 	private final List<RuleSet> rulesets;
-	private final Map<String, String> userData;
```
Comment (nedtwigg) [inline lib/src/diktat/java/com/diffplug/spotless/glue/diktat/DiktatFormatterFunc.java:36]: So `userData` is nuked in `1.2.0`, and anybody who had been using that should stop? If that's the case, that's okay, but it should be clearly noted in the changelogs, and `userData` should be marked deprecated and should throw an error in the Gradle and Maven DSL.

**Review by nedtwigg — CHANGES_REQUESTED** [review]
If the code now requires `1.2.0` or greater, that's fine, but:

- it needs to be marked in the changelogs
- `DiktatStep.create` needs to parse the version number, and proactively throw an error if less than 1.2.0 ([`BadSemver`](https://github.com/diffplug/spotless/blob/main/lib/src/main/java/com/diffplug/spotless/kotlin/BadSemver.java) should make this pretty easy.)

**Comment by nedtwigg** [comment]
(Also you need to run `gradlew spotlessApply`)

File: lib/src/diktat/java/com/diffplug/spotless/glue/diktat/DiktatFormatterFunc.java (hunk 1 above)
Comment (nulls) [reply lib/src/diktat/java/com/diffplug/spotless/glue/diktat/DiktatFormatterFunc.java:36]: yes, changing of `usedData` was prohibited, but diktat uses it only for internal logic (it was not supposed to be set by client).
will add note to the changelogs and add errors to plugins

Comment (nulls) [reply lib/src/diktat/java/com/diffplug/spotless/glue/diktat/DiktatFormatterFunc.java:36]: @nedtwigg, actually, I don't see calls of method from `DiktatStep` with `userData` parameter.
I only see calls:
1. `com.diffplug.spotless.kotlin.DiktatStep#create(java.lang.String, com.diffplug.spotless.Provisioner, com.diffplug.spotless.FileSignature)`:
    * `com.diffplug.gradle.spotless.KotlinExtension.DiktatFormatExtension#createStep` 
    * `com.diffplug.spotless.maven.kotlin.Diktat#newFormatterStep`
2. `com.diffplug.spotless.kotlin.DiktatStep#createForScript`: 
    * `com.diffplug.gradle.spotless.KotlinGradleExtension.DiktatFormatExtension#createStep`


So, it's safe to remove `userData`

**Review by nedtwigg — APPROVED** [review]

**Comment by nedtwigg** [comment]
Correct, Spotless classloaders are isolated. BTW, this PR bumps diktat to 1.2.1, which seems to not exist. I'm going to downgrade this to 1.2.0

https://search.maven.org/artifact/org.cqfn.diktat/diktat-rules

**Comment by nedtwigg** [comment]
Released in `plugin-gradle 6.8.0` and `plugin-maven 2.23.0`.

**Pull request merged**